targets/csharp/source/source/PlayFabGameServerInternal.cs

```diff
@@ -0,0 +1,263 @@
+﻿////////////////////////////////////////////////
+// Copyright (C) Microsoft. All rights reserved.
+////////////////////////////////////////////////
+
+// GSDK Only has the following supported frameworks for now
+#if NETSTANDARD2_0 || NETCOREAPP2_1
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PlayFab.GSDK
+{
+    internal class GameServerInternalSdk : IDisposable
+    {
+        private string _overrideConfigFileName;
+        private GameState _state;
+
+        private Task _heartbeatTask;
+        private IGameServerHttpClient _webClient;
+        private bool _heartbeatRunning;
+        private DateTime _cachedScheduleMaintDate;
+        private ManualResetEvent _heartbeatDoneEvent = new ManualResetEvent(false);
+        private ManualResetEvent _signalHeartbeatEvent = new ManualResetEvent(false);
+        private bool _shouldWriteDebugLogs;
+
+        internal GameServerConfiguration Configuration { get; private set; }
+
+        public ManualResetEvent TransitionToActiveEvent { get; set; }
+        public ILogger Logger { get; private set; }
+        public GameState State
+        {
+            get { return _state; }
+            set
+            {
+                if (_state != value)
+                {
+                    _state = value;
+                    _signalHeartbeatEvent.Set();
+                }
+            }
+        }
+
+        public IList<GameServerConnectedPlayer> ConnectedPlayers { get; set; }
+
+        public IList<string> InitialPlayers { get; set; }
+
+        // Keep an instance of the callbacks around so they don't get garbage collected
+        public Action ShutdownCallback { get; set; }
+        public Func<bool> HealthCallback { get; set; }
+        public Action<DateTimeOffset> MaintenanceCallback { get; set; }
+
+        public GameServerInternalSdk(string configFileName = null)
+        {
+            _overrideConfigFileName = configFileName;
+            ConnectedPlayers = new List<GameServerConnectedPlayer>();
+            InitialPlayers = new List<string>();
+            TransitionToActiveEvent = new ManualResetEvent(false);
+        }
+
+        public Task StartAsync(bool shouldWriteDebugLogs = false)
```

Comment:
Add a workitem to make this not a TAsk. There is nothing here that needs this to be a task. That way the other GSDK becomes cleaner (we don't need the .Wait everywhere)

Reply:
[https://dev.azure.com/playfab/PlayFab/_workitems/edit/27055](https://dev.azure.com/playfab/PlayFab/_workitems/edit/27055)

---
In reply to: [256592056](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r256592056) [](ancestors = 256592056)